File: backend/planit/src/main/java/pt/iade/planit/entity/Event.java
package pt.iade.planit.entity;

import jakarta.persistence.*;
import lombok.*;

import java.time.LocalDateTime;

@Entity
@Table(name = "Event")
@Data
@NoArgsConstructor
@AllArgsConstructor
public class Event {

    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    private Integer id;

    @Column(nullable = false)
    private String title;

    @Column
    private String description;

    @Column(nullable = false)
    private LocalDateTime date;

    @Column(name = "photo_url")
    private String photoUrl;

    @ManyToOne(optional = false)
    @JoinColumn(name = "user_id", nullable = false)
    @ToString.Exclude
    @EqualsAndHashCode.Exclude
    private User user;

    @ManyToOne
    @JoinColumn(name = "location_id")
    @ToString.Exclude
    @EqualsAndHashCode.Exclude
    private Location location;
}

File: android/app/src/main/java/pt/iade/planit/api/Event.kt
package pt.iade.planit.api

data class Event(
    val id: Int? = null,
    val userId: Int,
    val title: String,
    val description: String,
    val date: String,
    val photoUrl: String? = null,
    val latitude: Double,
    val longitude: Double,
    val address: String
)
